### PR TITLE
Group by list card view

### PR DESCRIPTION
**Pull request by phamann**
This is a large but ultimately quite simple PR to clean up the lists UI. As discussed, this changes the all lists view to be  part of the saved articles page, exposed via a `Group by: time | lists` toggle.

- Refactor `readAll` controller to fetch content for each list.
- Refactor `topic-card` layout and view decorator to allow for list cards.
- Create new `list card` library to model the list and content data into the correct view model.
- Refactor `toolbar` template to allow for multiple partials
- Add toolbar to existing list page for `delete` and `share` functionality.

#### New saved articles list view and toolbar
google chrome

#### New list view toolbar
google chrome

/cc @Financial-Times/next-myft

### CODE REVIEW

File: server/controllers/lists/readAll.js [hunk 1]
```diff
@@ -1,22 +1,49 @@
...
+const layouts = require('../../lib/layouts');
 const getLists = require('../../lib/lists');
+const getListContent = require('../../lib/fetch-list-content');
+const getLegacyContent = require('../../lib/get-legacy-content');
+const renderStream = require('../../lib/render-stream');
 
-module.exports = function(req, res, next) {
+const pluck = (prop, collection) => collection.map(c => c[prop]);
+const getContent = (flags) => {
+	return (uuid) => {
```
Comment (laurieboyes) [inline server/controllers/lists/readAll.js]: considering sticking one of these in our linting to move the burden of pedantry from us to the build pipeline: http://eslint.org/docs/rules/arrow-parens

Comment (laurieboyes) [inline server/controllers/lists/readAll.js]: it's interesting as well that they recommend having the parentheses all the time

Comment (keirog) [inline server/controllers/lists/readAll.js]: Would be nice if we could get nbt warnings as PR comments

Comment (laurieboyes) [inline server/controllers/lists/readAll.js]: If a PR or a branch build, parse eslint output (???) and fire off one of these: https://developer.github.com/v3/repos/comments/#create-a-commit-comment

File: views/partials/toolbars/list.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+							class="n-myft-ui n-myft-ui__button myft-lists__item__action js-share-list o--if-js"
+							data-trackable="share-list"
+							data-list-url="{{publicUrl}}"
+							{{#unless isPublic}}disabled{{/unless}}
+							>
+						Share
+					</button>
+
+					<form class="myft-lists__item__action myft-ui myft-ui--prefer" method="POST"
+						action="/__myft/api/core/list/{{uuid}}?method=put">
```
Comment (laurieboyes) [inline views/partials/toolbars/list.html]: we don't seem to have `uuid` in scope, this is just going to __myft/api/core/list/?method=put

File: views/partials/toolbars/list.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<div class="myft-toolbar">
+	<div class="o-grid-container">
+		<div class="o-grid-row">
+			<div data-o-grid-colspan="12">
+					<form class="n-myft-ui n-myft-ui--delete"
+						  method="POST"
+						  action="/__myft/api/core/list/{{uuid}}?method=delete"
```
Comment (laurieboyes) [inline views/partials/toolbars/list.html]: see below

**Comment by laurieboyes** [comment]
Toolbar in single list page core experience a bit funky
screen shot 2015-12-07 at 11 19 01

File: views/partials/toolbars/list.html (hunk 2 above)
Comment (laurieboyes) [inline views/partials/toolbars/list.html]: additionally, this redirects back to lists page (as opposed to page for this specific list)

**Pull request merged**